symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/StreamService.java

```diff
@@ -281,19 +414,64 @@ public V2AdminStreamList listStreamsAdmin(V2AdminStreamFilter filter) {
    * @return List of member in the stream with the given stream id.
    * @see <a href="https://developers.symphony.com/restapi/reference#stream-members">Stream Members</a>
    */
-  public V2MembershipList listStreamMembers(String streamId) {
+  public V2MembershipList listStreamMembers(@Nonnull String streamId) {
     return executeAndRetry("listStreamMembers",
         () -> streamsApi.v1AdminStreamIdMembershipListGet(streamId, authSession.getSessionToken(), null, null));
   }
 
+  /**
+   * List the current members of an existing stream.
+   * The stream can be of type IM, MIM, or ROOM.
+   *
+   * @param streamId The stream id
+   * @param pagination  The skip and limit for pagination.
+   * @return List of member in the stream with the given stream id.
+   * @see <a href="https://developers.symphony.com/restapi/reference#stream-members">Stream Members</a>
+   */
+  public V2MembershipList listStreamMembers(@Nonnull String streamId, @Nullable PaginationAttribute pagination) {
+    if (pagination != null) {
+      return executeAndRetry("listStreamMembers",
+          () -> streamsApi.v1AdminStreamIdMembershipListGet(streamId, authSession.getSessionToken(), pagination.skip(),
+              pagination.limit()));
+    }
+    return listStreamMembers(streamId);
+  }
+
+  /**
+   * List the current members of an existing room and return in a {@link java.util.stream.Stream}.
+   * The stream can be of type IM, MIM, or ROOM.
+   *
+   * @param streamId The stream id
+   * @return A {@link java.util.stream.Stream} of members in the stream with the given stream id.
+   * @see <a href="https://developers.symphony.com/restapi/reference#stream-members">Stream Members</a>
+   */
+  public java.util.stream.Stream<V2MemberInfo> listStreamMembersInStream(@Nonnull String streamId) {
+    return listStreamMembersInStream(streamId, null);
+  }
+
+  /**
+   * List the current members of an existing room and return in a {@link java.util.stream.Stream}.
+   * The stream can be of type IM, MIM, or ROOM.
+   *
+   * @param streamId    The stream id
+   * @param pagination  The chunkSize and totalSize for stream pagination with default value equal 100.
+   * @return A {@link java.util.stream.Stream} of members in the stream with the given stream id.
+   * @see <a href="https://developers.symphony.com/restapi/reference#stream-members">Stream Members</a>
+   */
+  public java.util.stream.Stream<V2MemberInfo> listStreamMembersInStream(@Nonnull String streamId, @Nullable StreamPaginationAttribute pagination) {
+    PaginatedApi<V2MemberInfo> api = (offset, limit) -> listStreamMembers(streamId, new PaginationAttribute(offset, limit)).getMembers();
+    pagination = PaginatedService.getValidPaginationAttribute(pagination, 100, 100);
```

Comment:
the default values feel like magic constants here, maybe we could at least move them to constants

Reply:
it depends on each api, i.e: for listMembers, the default value is 100, but 50 for listUserStreams. Or we can force the user using the same default value for all methods?